### PR TITLE
test(components): Unit tests for sagemaker components

### PR DESCRIPTION
**Pull request by ananth102**
**Description of your changes:**

Unit testing for common runtime and Training Job.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            [{"a": 1}, {"b": 2}], par.yaml_or_json_list('[ {"a": 1}, {"b": 2} ]')
+        )
+
+        with self.assertRaises(argparse.ArgumentTypeError):
+            par.yaml_or_json_list("{a: 1}")
+
+        self.assertEqual(
+            [{"a": 1}, {"b": 2}],
+            par.yaml_or_json_list(
+                """
```
Comment (ryansteakley) [inline components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py:21]: formatting here looks a little strange is this what we see after running black?

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        )
+
+    def test_str_to_bool(self):
+        self.assertTrue(par.str_to_bool("True"))
+        self.assertFalse(par.str_to_bool("False"))
+
+
+if __name__ == "__main__":
+    # unittest.main()
+    par.yaml_or_json_list("{a: 1}")
```
Comment (ryansteakley) [inline components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py:50]: Why is unittest.main() commented out and we have this extra call to `yaml_or_json_list` is this an artifact of testing?

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_sagemaker_component_v2.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+            ) as mock_k8s_context:
+                mock_k8s_context.return_value = (
+                    " ",
+                    {"context": {"namespace": "default-ns"}},
+                )
+                self.component._get_current_namespace()
+                self.assertEqual("default-ns", self.component._get_current_namespace())
+
+    @patch.object(SageMakerComponent, "_get_k8s_api_client", MagicMock())
+    def test_wait_resource_consumed_by_controller(self):
```
Comment (ryansteakley) [inline components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_sagemaker_component_v2.py:280]: nit: This is a general comment for several of the tests in this file, looks like several of the methods that are tested here have varying return values i.e(True/False, name-spaced or not). In some of these tests its a little hard to read what is going on and quickly discern what the test is about. Maybe its worth splitting some of these longer tests into seperate subtests for readibility sake

File: components/aws/sagemaker/tests/unit_tests/tests/TrainingJob/test_TrainingJob_spec.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        "{'instanceCount': 1, 'instanceType': 'ml.m4.xlarge', 'volumeSizeInGB': 5}",
+        "--role_arn",
+        "arn:aws:iam::123456789012:role/ack-sagemaker-execution-role",
+        "--stopping_condition",
+        "{'maxRuntimeInSeconds': 86400}",
+        "--training_job_name",
+        "kfp-ack-training-job-99999",
+    ]
+
+    def test_minimum_required_args(self):
```
Comment (ryansteakley) [inline components/aws/sagemaker/tests/unit_tests/tests/TrainingJob/test_TrainingJob_spec.py:28]: Would adding a test with incorrect inputs to verify the exception is raised be a large effort?

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_sagemaker_component_v2.py (hunk 3 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_sagemaker_component_v2.py:280]: Will split and add comments.

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py (hunk 1 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py:21]: Yea its the same

File: components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py (hunk 2 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/unit_tests/tests/commonv2/test_spec_input_parsers_v2.py]: fixed.

File: components/aws/sagemaker/tests/unit_tests/tests/TrainingJob/test_TrainingJob_spec.py (hunk 4 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/unit_tests/tests/TrainingJob/test_TrainingJob_spec.py:31]: done

**Review by ryansteakley — APPROVED** [review]

**Pull request closed without merging**